codegen/module.go

```diff
@@ -269,6 +269,54 @@ func (system *ModuleSystem) ResolveModules(
 					append(resolvedDependencies, dependencyInstance)
 			}
 		}
+
+		// Resolve the class dependencies
+		for _, classInstance := range classInstances {
```

Comment:
Yes we have this loop twice because the current class instances need to be fully resolved before you can resolve the dependencies (because clients can have peer dependencies, for example). So basically you have to create all possible dependency ModuleInstance objects before you can populate the ResolvedDependencies. I don't think this is incorrect.